test/docker/prepare-ci.sh

```
@@ -0,0 +1,127 @@
+    local dockerfile="${3:-Dockerfile}"
+
+    if image_exists "$tag"; then
+        echo "✓ Local image $tag already exists"
+    else
+        echo "🔨 Building $tag from $context..."
+        docker build -t "$tag" -f "$context/$dockerfile" "$context"
+    fi
+}
+
+# Ensure Docker is available
+if ! command -v docker &> /dev/null; then
```

Shellcheck (0.10.0)

[warning] In POSIX sh, &> is undefined.

(SC3020)